### PR TITLE
Vulkan Example

### PR DESCRIPTION
**Pull request by Loftilus**
- This example needs an update version of GLFW.
- It currently uses the Vulkan headers from the GLFW repo. We can maybe put that in the libs folder?
- To keep the code compact i decided to not properly setup Vulkan in main. I don't know if this will be an issue.

### CODE REVIEW

File: examples/vulkan_example/main.cpp [hunk 1]
```diff
@@ -0,0 +1,533 @@
...
+    VkResult err;
+    while(true){
+        err = vkWaitForFences(g_Device, 1, &g_Fence[g_FrameIndex], VK_TRUE, 100);
+        if(err == VK_SUCCESS) break;
+        if(err == VK_TIMEOUT) continue;
+        check_vk_result(err);
+    }
+    {
+        vkDestroySemaphore(g_Device, g_Semaphore[g_FrameIndex], g_Allocator);
+        VkSemaphoreCreateInfo info = {};
```
Comment (ratchetfreak) [inline examples/vulkan_example/main.cpp]: You shouldn't need to delete and recreate the semaphores.

**Comment by ocornut** [comment]
Thanks. Will look at it when I have time.
Also see https://github.com/ocornut/imgui/issues/531 "Add Vulkan rendering backend example" and https://github.com/tambry/ImGuiVulkanRenderer

**Comment by fire** [comment]
There is still interest in this. Hopefully it can be merged soon.

**Comment by ocornut** [comment]
Unfortunately it takes time to review those samples! You can probably merge them locally yourself and use the code already, nothing stops you from doing that :)

This looks mostly good but still needs some cleanup, if a PR comes not following the style-syntax-coding convention of other examples very precisely, my heuristic tells me that it's coming with other issues. 

So as I understand the two shaders are built from source into SPV using the shell script and only the SPV is needed to run the example, not the shader sources?
This would probably need a readme to explain that flow, and a makefile. I'll look into it eventually.

**Comment by ocornut** [comment]
Merged (applied coding style changes but haven't changed the actual behavior apart from a few minor changes applied to other examples since this one was forked).

The situation with building examples is a total mess and we should aim to provide more organized build scripts eventually (perhaps using genie).

**Pull request merged**